Remove `public` from the extension declaration and add it to each declaration in the extension

Matejkob opened the pull request.

Matejkob commented on Sources/SwiftCompilerPluginMessageHandling/PluginMessageCompatibility.swift:
```diff
@@ -11,8 +11,8 @@
 /// Old compiler might send '.declaration' as "freeStandingDeclaration".
-@_spi(PluginMessage) public extension PluginMessage.MacroRole {
-  init(from decoder: Decoder) throws {
+@_spi(PluginMessage) extension PluginMessage.MacroRole {
```
Shouldn't the `@_spi` attribute be moved as well?

ahoppen replied:
Yes, could you file an issue for that at https://github.com/apple/swift-format?

Matejkob replied:
https://github.com/apple/swift-format/issues/714

ahoppen commented:
I don’t know with whom I had a conversation about this at some point but I think that `private` and `fileprivate` extensions make sense to have. That way if you read `private extension`, you don’t need to worry about anything leaking out into the rest of the module. `public` extensions are what I agree can be a source of unintended behavior. Could you maybe:
- Update this PR to remove all `public` extensions but keep `private` and `fileprivate`
- File an issue on https://github.com/apple/swift-format to add a rule to only

Matejkob commented:
> I don’t know with whom I had a conversation about this at some point but I think that `private` and `fileprivate` extensions make sense to have. That way if you read `private extension`, you don’t need to worry about anything leaking out into the rest of the module. `public` extensions are what I agree can be a source of unintended behavior. Could you maybe:
> 
> * Update this PR to remove all `public` extensions but keep `private` and `fileprivate`
> * File an issue on https://github.com/apple/swift-format to add a rule to only disallows `public extension`?

I've removed the `swift-format` rule and moved the `public` access level attribute manually.

ahoppen commented on Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:
```diff
@@ -69,8 +69,10 @@ public protocol HasTrailingCodeBlock: WithCodeBlockSyntax {
-public extension HasTrailingCodeBlock where Self: StmtSyntaxProtocol {
-  init(_ header: SyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () throws -> CodeBlockItemListSyntax) throws {
+extension HasTrailingCodeBlock where Self: StmtSyntaxProtocol {
+  public init(_ header: SyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () throws -> CodeBlockItemListSyntax)
+    throws
```
Could you format this to break at the arguments?

```suggestion
  public init(
    _ header: SyntaxNodeString, 
    @CodeBlockItemListBuilder bodyBuilder: () throws -> CodeBlockItemListSyntax
  ) throws
```

ahoppen commented on Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:
```diff
@@ -117,7 +119,9 @@ public extension WithOptionalCodeBlockSyntax where Self: DeclSyntaxProtocol {
   ///
   /// Throws an error if `header` defines a different node type than the type the initializer is called on. E.g. if calling `try FunctionDeclSyntax("init") {}`
-  init(_ header: SyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () throws -> CodeBlockItemListSyntax) throws {
+  public init(_ header: SyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () throws -> CodeBlockItemListSyntax)
+    throws
```
```suggestion
  public init(
    _ header: SyntaxNodeString, 
    @CodeBlockItemListBuilder bodyBuilder: () throws -> CodeBlockItemListSyntax
  ) throws
```

ahoppen commented on Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift:
```diff
@@ -362,7 +366,9 @@ public extension VariableDeclSyntax {
   ///
   /// Throws an error if `header` does not start a variable declaration. E.g. if calling `try VariableDeclSyntax("func foo") {}`
-  init(_ header: SyntaxNodeString, @CodeBlockItemListBuilder accessor: () throws -> CodeBlockItemListSyntax) throws {
+  public init(_ header: SyntaxNodeString, @CodeBlockItemListBuilder accessor: () throws -> CodeBlockItemListSyntax)
+    throws
```
```suggestion
  public init(
    _ header: SyntaxNodeString, 
    @CodeBlockItemListBuilder accessor: () throws -> CodeBlockItemListSyntax
  ) throws
```

ahoppen reviewed (commented):
Thank you 🙏🏽  Just a few formatting suggestions, otherwise looks good to me 👍🏽

ahoppen reviewed (approved):
Thanks 👍🏽

ahoppen commented:
@swift-ci Please test

ahoppen commented:
@swift-ci Please test Windows

ahoppen commented:
Could you also create a cherry-pick PR of this to the `release/6.0` branch? I think it could make future cherry-picks easier.
